.github/workflows/pre-commit.yml

```
@@ -1,33 +1,22 @@
-name: Run pre-commit job
+name: lint
 
 on:
- push:
+  # trigger on pushes to any branch, but not main
+  push:
+    branches-ignore:
+      - master
```

Did you mean `main`?
```suggestion
      - main
```
Note that admin can still force push to main.